program hard fails when any of the following files are missing: server_config.json, github-keys.json, configuration.json, base.js

## Hi! Thank you for getting back to us! <3

### Is this a feature request or a bug report? 
* Please choose one of the labels on the right side and tag your issue :)

### What is the current behaviour?
New developer `git clone REPO`, `npm i`, `npm run start`, and is greeted with error message

### What is the expected or desired behavior?
Throw a warning rather than an error. Also show the warning in UI in `/` and/or `/data`.

### Version information (for bug reports)
Chrome 61.0.3163.100
Windows 10
```
node --version && npm --version
```
node --version: v6.10.2
npm --version; 3.10.10

`npm run start` log:
`
> microdraw@0.0.1 start C:\git projects\microdraw20\m2\microdraw
> node ./bin/www
fs.js:641
  return binding.open(pathModule._makeLong(path), stringToFlags(flag
                 ^
Error: ENOENT: no such file or directory, open 'C:\git projects\micr
    at Error (native)
    at Object.fs.openSync (fs.js:641:18)
    at Object.fs.readFileSync (fs.js:509:33)
    at Object.<anonymous> (C:\git projects\microdraw20\m2\microdraw\
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
npm ERR! Windows_NT 10.0.15063
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Fil
" "start"
npm ERR! node v6.10.2
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! microdraw@0.0.1 start: `node ./bin/www`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the microdraw@0.0.1 start script 'node ./bin/www'
npm ERR! Make sure you have the latest version of node.js and npm in
npm ERR! If you do, this is most likely a problem with the microdraw
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./bin/www
npm ERR! You can get information on how to open an issue for this pr
npm ERR!     npm bugs microdraw
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls microdraw
npm ERR! There is likely additional logging output above.
npm ERR! Please include the following file with any support request:
npm ERR!     C:\git projects\microdraw20\m2\microdraw\npm-debug.log
`

--- a/app.js
+++ b/app.js
@@ -51,10 +51,22 @@ app.use(express.static(path.join(__dirname, 'public')));
 var session = require('express-session');
 var passport = require('passport');
 var GithubStrategy = require('passport-github').Strategy;
-passport.use(new GithubStrategy(
-    JSON.parse(fs.readFileSync(dirname + "/github-keys.json")),
-    function (accessToken, refreshToken, profile, done) { return done(null, profile); }
-));
+var warningGithubConfig = false;
+fs.readFile("github-keys.json","utf8",(err,githubKeys)=>{
+    if(err){
+        console.warn('github-keys.json does not exist. Users cannot be authenticated ...',err)
+        warningGithubConfig = true
+    }else{
+        try{
+            passport.use(new GithubStrategy( JSON.parse(githubKeys), 
+            (accessToken,refreshToken,profile,done)=>done(null,profile)))
+        }catch(e){
+            console.warn('parsing githubkey.json or passport.use(new GithubStrategy error',e)
+            warningGithubConfig = true
+        }
+    }
+})
+
 app.use(session({
     secret: "a mi no me gusta la sémola",
     resave: false,
@@ -134,7 +146,10 @@ app.get('/', function (req, res) { // /auth/github
     });
 });
 
-app.use('/data', require('./controller/data/'));
+app.use('/data', (req,res,next)=>{
+    req.warningGithubConfig = warningGithubConfig;
+    next();
+},require('./controller/data/'));
 
 // API routes
 app.get('/api', function (req, res) {

--- a/app.js
+++ b/app.js
@@ -52,20 +52,20 @@ var session = require('express-session');
 var passport = require('passport');
 var GithubStrategy = require('passport-github').Strategy;
 var warningGithubConfig = false;
-fs.readFile("github-keys.json","utf8",(err,githubKeys)=>{
-    if(err){
-        console.warn('github-keys.json does not exist. Users cannot be authenticated ...',err)
-        warningGithubConfig = true
+fs.readFile("github-keys.json", "utf8", (err, githubKeys) => {
+    if (err) {
+        console.warn('github-keys.json does not exist. Users would not be able to be authenticated ...', err);
+        warningGithubConfig = true;
     }else{
         try{
-            passport.use(new GithubStrategy( JSON.parse(githubKeys), 
-            (accessToken,refreshToken,profile,done)=>done(null,profile)))
-        }catch(e){
-            console.warn('parsing githubkey.json or passport.use(new GithubStrategy error',e)
-            warningGithubConfig = true
+            passport.use(new GithubStrategy( JSON.parse(githubKeys),
+            (accessToken, refreshToken, profile, done) => done(null, profile)));
+        }catch (e) {
+            console.warn('parsing githubkey.json or passport.use(new GithubStrategy) error', e);
+            warningGithubConfig = true;
         }
     }
-})
+});
 
 app.use(session({
     secret: "a mi no me gusta la sémola",
@@ -146,10 +146,10 @@ app.get('/', function (req, res) { // /auth/github
     });
 });
 
-app.use('/data', (req,res,next)=>{
+app.use('/data', (req, res, next) => {
     req.warningGithubConfig = warningGithubConfig;
     next();
-},require('./controller/data/'));
+}, require('./controller/data/'));
 
 // API routes
 app.get('/api', function (req, res) {

--- a/controller/data/data.controller.js
+++ b/controller/data/data.controller.js
@@ -4,7 +4,8 @@ function data(req, res) {
     var { source: source } = req.query; // eslint-disable-line no-unused-vars, no-useless-rename
     var login = (req.isAuthenticated()) ?
                 ("<a href='/user/" + req.user.username + "'>" + req.user.username + "</a> (<a href='/logout'>Log Out</a>)")
-                : ("<a href='/auth/github'>Log in with GitHub</a>");
+                : req.warningGithubConfig ? ("Github OAuth failed ... ")
+                    :("<a href='/auth/github'>Log in with GitHub</a>");
     var loggedUser = req.isAuthenticated()?req.user.username:"anonymous"; // eslint-disable-line no-unused-vars
 
     // store return path in case of login

